clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -324,6 +325,21 @@ void SYCL::fpga::BackendCompiler::ConstructJob(
       FPGADepFiles.push_back(II);
     else
       CmdArgs.push_back(C.getArgs().MakeArgString(Filename));
+    // Check for any AOCR input, if found use that as the project report name
+    StringRef Ext(llvm::sys::path::extension(Filename));
+    if (Ext.empty())
+      continue;
+    if (getToolChain().LookupTypeForExtension(Ext.drop_front()) ==
+        types::TY_FPGA_AOCR) {
+      // Keep the base of the .aocr file name.  Input file is a temporary,
+      // so we are stripping off the additional naming information for a
+      // cleaner name.  The length is the being stripped from the name is the
+      // added temporary string and the extension:  input_file-XXXXXX.aocr
+      SmallString<128> NameBase(Filename.substr(0, Filename.length() - 12));
```

Comment:
I would suggest something like this with the aim of guarding us humans from manual counting:
```suggestion
      // cleaner name. The length is the being stripped from the name is the
      // added temporary string and the extension:
      constexpr std::string_view SuffixFormat = "input_file-XXXXXX.aocr";
      SmallString<128> NameBase(Filename.substr(0, Filename.length() - SuffixFormat.length()));
```

Reply:
So yeah, perhaps just this. I don't know if our `StringRef` grants the same compile-time fanciness, but should still be fine for the minor purpose.
```suggestion
      // cleaner name. The length is the being stripped from the name is the
      // added temporary string and the extension:
      StringRef SuffixFormat("input_file-XXXXXX.aocr");
      SmallString<128> NameBase(Filename.substr(0, Filename.length() - SuffixFormat.size()));
```
EDIT: string_view is C++17, though... I'll update the comment if I find a synonym in C++11.

Reply:
I like it - done.